codex/stores/repostore.nim

```diff
@@ -144,18 +144,22 @@ method ensureExpiry*(
   if expiry <= 0:
     return failure(newException(ValueError, "Expiry timestamp must be larger then zero"))
 
-  let expiryKey = createBlockExpirationMetadataKey(cid)
-  let currentExpiry = await self.metaDs.get(expiryKey.tryGet)
+  without expiryKey =? createBlockExpirationMetadataKey(cid), err:
+    return failure(err)
 
-  if currentExpiry.isOk and expiry < (!currentExpiry).toSecondsSince1970:
-    trace "Current expiry is larger then the specified one, no action needed"
-    return success()
+  without currentExpiry =? await self.metaDs.get(expiryKey), err:
+    if err of DatastoreKeyNotFound:
+      error "No current expiry exists for the block"
```

Comment:
In this branch I think you should return:
`failure(newException(BlockNotFoundError, err.msg))`

Reply:
It is maybe bit long stretch, but agree